cmd/odo/odo.go

```diff
@@ -49,9 +48,11 @@ func main() {
 	// Override the logging level by the value (if set) by the ODO_LOG_LEVEL env
 	// The "-v" flag set on command line will take precedence over ODO_LOG_LEVEL env
 	v := flag.CommandLine.Lookup("v").Value.String()
-	// if the json flag is passed and is valid, we don't turn on ODO_LOG_LEVEL
-	jsonFlagValue := flag.CommandLine.Lookup("o").Value.String()
-	if level, ok := os.LookupEnv("ODO_LOG_LEVEL"); ok && v == "0" && strings.ToLower(jsonFlagValue) != "json" {
+
+	// if the output flag is set, we don't turn on ODO_LOG_LEVEL
+	outputFlag := pflag.Lookup("o")
+	hasFlagChanged := outputFlag != nil && outputFlag.Changed
+	if level, ok := os.LookupEnv("ODO_LOG_LEVEL"); ok && v == "0" && hasFlagChanged {
```

Comment:
Should the last condition be `!hasFlagChanged`?

Reply:
Ah, you're right! I totally missed the `!=` in the `strings.ToLower(jsonFlagValue) != "json"`